src/base/QXmppJingleIq.h

```diff
@@ -89,6 +90,66 @@ class QXMPP_EXPORT QXmppJingleRtpFeedbackInterval
     uint64_t m_value;
 };
 
+class QXMPP_EXPORT QXmppJingleRtpHeaderExtensionProperty
+{
+public:
+    enum Senders {
+        /// The initiator and the sender are allowed.
+        Both,
+        /// Only the initiator is allowed.
+        Initiator,
+        /// Only the responder is allowed.
+        Responder
+    };
+
+    QXmppJingleRtpHeaderExtensionProperty();
+
+    QXMPP_PRIVATE_DECLARE_RULE_OF_SIX(QXmppJingleRtpHeaderExtensionProperty)
+
+    uint32_t id() const;
+    void setId(uint32_t id);
+
+    QString uri() const;
+    void setUri(const QString &uri);
+
+    Senders senders() const;
+    void setSenders(Senders senders);
+
+    QVector<QXmppSdpParameter> parameters() const;
+    void setParameters(const QVector<QXmppSdpParameter> &parameters);
+
+    /// \cond
+    void parse(const QDomElement &element);
+    void toXml(QXmlStreamWriter *writer) const;
+    /// \endcond
+
+    static bool isJingleRtpHeaderExtensionProperty(const QDomElement &element);
+
+private:
+    QSharedDataPointer<QXmppJingleRtpHeaderExtensionPropertyPrivate> d;
+};
+
+class QXMPP_EXPORT QXmppJingleRtpHeaderExtensionMixing
+{
+public:
+    QXmppJingleRtpHeaderExtensionMixing();
+
+    QXMPP_PRIVATE_DECLARE_RULE_OF_SIX(QXmppJingleRtpHeaderExtensionMixing)
+
+    bool isAllowed() const;
+    void setAllowed(bool isAllowed);
+
+    /// \cond
+    void parse(const QDomElement &element);
+    void toXml(QXmlStreamWriter *writer) const;
+    /// \endcond
+
+    static bool isJingleRtpHeaderExtensionMixing(const QDomElement &element);
+
+private:
+    bool m_isAllowed = false;
+};
```

Comment:
If this is really for just one bool, I think an extra class with parse(), toXml() etc. could be too much here? Do you think it is a good idea?